src/interface/report/Results/Timeline/EnemyCasts.tsx

```diff
@@ -277,39 +281,37 @@ export const EnemyCastsTimeline = ({
             parser.fight.start_time,
             parser.fight.end_time,
             undefined,
-            "type in ('begincast', 'cast') and source.type in ('NPC', 'Boss') AND ability.id > 1 AND sourceID > -1",
+            `type in ('begincast', 'cast') and source.type in ('NPC', 'Boss') AND ability.id > 1 AND sourceID > -1 AND ability.id not in (${EXCLUDED_SPELLS.join(',')})`,
             40,
           )) as (NpcBeginCastEvent | NpcCastEvent)[];
+          const abilities = new Set();
+          for (const event of events) {
+            abilities.add(event.ability.guid);
+          }
           //This call grabs the damage events that friendly players took from NPCs
           const damageStuff = (await fetchEvents(
             parser.report.code,
             parser.fight.start_time,
             parser.fight.end_time,
             undefined,
-            "type = 'damage' AND source.type in ('NPC', 'Boss') AND ability.id > 1 AND source.id > 0 AND target.type = 'Player'",
+            `type = 'damage' AND source.type in ('NPC', 'Boss') AND ability.id > 1 AND source.id > 0 AND target.type = 'Player' AND missType not in ('immune', 'deflect', 'reflect', 'misfire', 'evade') AND ability.id in (${Array.from(abilities).join(',')})`,
```

Comment:
this has a small chance of breaking once live due to URL limitations. basically: if too many unique ability ids are included, this will fail to send the request because the URL will be too long. that happened with the `TimeOfNeed` module, but I *think* this will end up safe enough. I tested a couple keys + some raid bosses